liquibase-standard/src/main/java/liquibase/command/core/DbDocCommandStep.java

```diff
@@ -1,64 +1,142 @@
+            checkLiquibaseTables(false, databaseChangeLog, new Contexts(), new LabelExpression(), database);
+
+            databaseChangeLog.validate(database, contexts, labelExpression);
+
+            ChangeLogIterator logIterator = new ChangeLogIterator(databaseChangeLog, new DbmsChangeSetFilter(database));
+
+            DBDocVisitor visitor = new DBDocVisitor(database);
+            logIterator.run(visitor, new RuntimeEnvironment(database, contexts, labelExpression));
+
+            final PathHandlerFactory pathHandlerFactory = Scope.getCurrentScope().getSingleton(PathHandlerFactory.class);
+            Resource resource = pathHandlerFactory.getResource(outputDirectory);
+            visitor.writeHTML(resource, resourceAccessor, catalogAndSchemas);
+            resultsBuilder.addResult("statusCode", 0);
+        } catch (IOException e) {
+            resultsBuilder.addResult("statusCode", 1);
+            throw new LiquibaseException(e);
+        }
+    }
+
+    /**
+     * Finds the appropriate catalog and schema for the query
+     *
+     * @param schemas             the comma list of schemas input to the command via the command framework or cli
+     * @param catalogAndSchemaArg the list of catalog and schema from Main or Liquibase classes. This will be used if present.
+     * @param database            the database to find the catalog and schema against
+     * @return the catalog and schema array to use
+     */
+    private CatalogAndSchema[] getCatalogAndSchema(String schemas, CatalogAndSchema[] catalogAndSchemaArg, Database database) {
+        if (catalogAndSchemaArg != null) {
+            return catalogAndSchemaArg;
+        } else if (schemas != null) {
+            List<CatalogAndSchema> schemaList = new ArrayList<>();
+            for (String schema : schemas.split(",")) {
+                schemaList.add(new CatalogAndSchema(null, schema).customize(database));
+            }
+            return schemaList.toArray(new CatalogAndSchema[0]);
+        }
+        return new CatalogAndSchema[]{new CatalogAndSchema(null, null)};
+    }
+
+    // Unsure if this method is still needed with the refactoring, but I'm leaving it in now to keep the logic the same.
+    public void checkLiquibaseTables(boolean updateExistingNullChecksums, DatabaseChangeLog databaseChangeLog,
+                                     Contexts contexts, LabelExpression labelExpression, Database database) throws LiquibaseException {
+        ChangeLogHistoryService changeLogHistoryService = ChangeLogHistoryServiceFactory.getInstance().getChangeLogService(database);
+        changeLogHistoryService.init();
+        if (updateExistingNullChecksums) {
+            changeLogHistoryService.upgradeChecksums(databaseChangeLog, contexts, labelExpression);
+        }
+        LockServiceFactory.getInstance().getLockService(database).init();
+    }
```

Comment:
Let me know if you think it's safe to actually get rid of this method/call or not. We already make a check (without updating checksums) when we use the DbUrlCommandStep. (This is called on line 92)